src/pytorch_lightning/loops/fit_loop.py

```diff
@@ -50,10 +50,10 @@ class FitLoop(Loop[None]):
     def __init__(
         self,
         min_epochs: int = 0,
-        max_epochs: int = 1000,
+        max_epochs: Optional[int] = None,
```

Comment:
@rohitgr7 Why this change? It's cleaner if the loop takes the argument already validated (converted to int). We should always try to avoid Optionals

Reply:
the issue is max_epochs doesn't do anything for `trainer.test/validate/predict` but we initialize the fit_loop in the trainer constructor. We need to ensure it gets updated and that warning should be raised during `.fit` only. Due to this, I don't see a possibility of setting it to an int before the call.